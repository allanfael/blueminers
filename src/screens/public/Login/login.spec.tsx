import React from 'react'
import { FormProvider, useForm } from 'react-hook-form'
import { render, screen } from '@testing-library/react-native'

import { useLogin } from './useLogin'
import { Login } from '.'

jest.mock('./useLogin', () => ({
  useLogin: jest.fn(),
}))

jest.mock('react-native-keyboard-aware-scroll-view', () => {
  const KeyboardAwareScrollView = ({ children }: any) => {
    return <>{children}</>
  }
  return { KeyboardAwareScrollView }
})

jest.mock('../../../store/user', () => ({
  userStore: {
    save: jest.fn(),
  },
}))

describe('<Login />', () => {
  it('should render Login correctly', async () => {
    ;(useLogin as jest.MockedFunction<typeof useLogin>).mockReturnValue({
      onSubmit: jest.fn(),
      form: {
        formState: {
          isSubmitting: false,
          errors: {
            email: {
              message: '',
            },
            password: {
              message: '',
            },
          },
        } as any,
        handleSubmit: jest.fn(),
      } as any,
    })

    render(<Login />)

    expect(screen.toJSON()).toMatchSnapshot()
  })

  it('should render with email and password correctly', async () => {
    ;(useLogin as jest.MockedFunction<typeof useLogin>).mockReturnValue({
      onSubmit: jest.fn(),
      form: {
        formState: {
          isSubmitting: false,
          errors: {
            email: {
              message: '',
            },
            password: {
              message: '',
            },
          },
        },
        handleSubmit: jest.fn(),
      } as any,
    })

    const methods = useForm({
      defaultValues: { email: '', password: '' },
    })

    methods.setValue('email', 'test@email')
    methods.setValue('password', '123123')

    render(
      <FormProvider {...methods}>
        <Login />
      </FormProvider>,
    )

    const email = await screen.getByTestId('login-email')
    const password = await screen.getByTestId('login-password')

    expect(email.props.value).toBe('test@email')
    expect(password.props.value).toBe('123123')
  })

  it('should render submit button disabled', async () => {
    ;(useLogin as jest.MockedFunction<typeof useLogin>).mockReturnValue({
      onSubmit: jest.fn(),
      form: {
        formState: {
          isSubmitting: true,
          errors: {
            email: {
              message: '',
            },
            password: {
              message: '',
            },
          },
        },
        handleSubmit: jest.fn(),
      } as any,
    })

    const methods = useForm({
      defaultValues: { email: '', password: '' },
    })

    methods.setValue('email', 'test@email')
    methods.setValue('password', '123123')

    render(
      <FormProvider {...methods}>
        <Login />
      </FormProvider>,
    )

    const button = await screen.findByTestId('login-button')

    expect(button.props.enabled).toBeFalsy()
  })
})
