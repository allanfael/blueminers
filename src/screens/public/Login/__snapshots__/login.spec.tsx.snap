// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render Login correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    [
      undefined,
      undefined,
      {
        "paddingBottom": 0,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 20,
      },
    ]
  }
  showsVerticalScrollIndicator={false}
>
  <View>
    <Text
      style={
        {
          "color": "#000",
          "fontFamily": "Muli_600SemiBold",
          "fontSize": 28,
          "lineHeight": 56,
        }
      }
    >
      Bem vindo
    </Text>
    <Text
      style={
        {
          "color": "#666",
          "fontFamily": "Muli_600SemiBold",
          "fontSize": 16,
          "lineHeight": 18,
        }
      }
    >
      Entre para continuar
    </Text>
    <View
      style={
        [
          undefined,
          undefined,
        ]
      }
    >
      <Text
        style={
          {
            "color": "#666",
            "fontFamily": "Muli_600SemiBold",
            "fontSize": 14,
            "lineHeight": 14,
          }
        }
      >
        E-mail
      </Text>
      <TextInput
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Digite seu e-mail"
        placeholderTextColor="#666"
        style={
          [
            undefined,
            {
              "backgroundColor": "#e0e0e0",
              "color": "#000",
            },
          ]
        }
        testID="login-email"
        textContentType="emailAddress"
        value=""
      />
    </View>
    <View
      style={
        [
          undefined,
          undefined,
        ]
      }
    >
      <Text
        style={
          {
            "color": "#666",
            "fontFamily": "Muli_600SemiBold",
            "fontSize": 14,
            "lineHeight": 14,
          }
        }
      >
        Senha
      </Text>
      <TextInput
        onChangeText={[Function]}
        placeholder="Digite sua senha"
        placeholderTextColor="#666"
        returnKeyLabel="Entrar"
        returnKeyType="go"
        secureTextEntry={true}
        style={
          [
            undefined,
            {
              "backgroundColor": "#e0e0e0",
              "color": "#000",
            },
          ]
        }
        testID="login-password"
        value=""
      />
    </View>
    <RNGestureHandlerButton
      activeOpacity={0.105}
      collapsable={false}
      delayLongPress={600}
      enabled={true}
      handlerTag={1}
      handlerType="NativeViewGestureHandler"
      onActiveStateChange={[Function]}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      style={
        {
          "backgroundColor": "#1f35d9",
        }
      }
      testID="login-button"
      underlayColor="black"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "black",
            "borderBottomLeftRadius": undefined,
            "borderBottomRightRadius": undefined,
            "borderRadius": undefined,
            "borderTopLeftRadius": undefined,
            "borderTopRightRadius": undefined,
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <Text
        style={
          {
            "color": "#fff",
            "fontFamily": "Muli_700Bold",
            "fontSize": 16,
            "lineHeight": 18,
          }
        }
      >
        Entrar
      </Text>
    </RNGestureHandlerButton>
  </View>
</RCTScrollView>
`;
